### DISCUSSION TITLE
[hooks]: useFormContext cause page error

### DISCUSSION THREAD

**Discussion by vaynevayne** [mantinedev/mantine#4275] [Q&A]
### What package has an issue

@mantine/core

### Describe the bug

<img width="1269" alt="image" src="https://github.com/mantinedev/mantine/assets/124666577/5af0ee9d-1625-4752-8dff-a5abdb11dba4">

<img width="790" alt="image" src="https://github.com/mantinedev/mantine/assets/124666577/8245e939-8f53-4c32-99b0-5b5a50bfdcc1">

I have a DatePicker component, and sometimes it's used in form. I need form data, which is good.
But sometimes he doesn't use it in form, and I don't need form data, but the useFormContext code still exists, which leads to an error on the page. I hope it's not throw Error, just console.

or, Is there a way I can catch an error that will not trigger reactErrorBoundry? 

### What version of @mantine/hooks page do you have in package.json?

latest

### Do you know how to fix the issue

None

### Are you willing to participate in fixing this issue and create a pull request with the fix

None

**Comment by vaynevayne** [comment]
https://github.com/react-hook-form/react-hook-form/blob/master/src/useFormContext.tsx#L37
react-hook-form No context judgment.，i think use console.warn is better

**Comment by rtivital** [comment]
`useFormContext` depends on Form context, if it is used outside of the form context it cannot return any associated values.

**Reply by vaynevayne** [reply]
In fact, I don't need the value of context when using outside context.

```jsx
// It is easy to be compatible with the two environments.
const form = useFormContext()
let name = form ? form.name : '_name'
```
But this code completely prevents me from using the component outside of context. Do I have to make a complete copy of the component?

**Reply by rtivital** [reply]
If you do not need context, then call `useFormContext` inside `if`

**Reply by vaynevayne** [reply]
useFormContext  iis import  from  mantine

**Reply by vaynevayne** [reply]
your means `if() useFormContext()`? but  limit Hooks used in if

**Reply by vaynevayne** [reply]
I would like to make sure that this problem will be dealt with. @rtivital
